src/controller/debtor-controller.ts

```diff
@@ -0,0 +1,233 @@
+    this.logger.trace('Get fine handout event', id, 'by', req.token.user);
+
+    try {
+      res.json(await DebtorService.getSingleFineHandoutEvent(Number.parseInt(id, 10)));
+    } catch (error) {
+      this.logger.error('Could not return fine handout event:', error);
+      res.status(500).json('Internal server error.');
+    }
+  }
+
+  /**
+   * Delete a fine
+   * @route DELETE /fines/{id}
+   * @group debtors - Operations of the debtor controller
+   * @security JWT
+   * @param {integer} id.path.required - The id of the fine which should be deleted
+   * @returns {} 204 - Success
+   * @returns {string} 400 - Validation error
+   * @returns {string} 500 - Internal server error
+   */
+  public async deleteFine(req: RequestWithToken, res: Response): Promise<void> {
+    const { id } = req.params;
+    this.logger.trace('Delete fine', id, 'by', req.token.user);
+
+    try {
+      const parsedId = Number.parseInt(id, 10);
+      const fine = await Fine.findOne({ where: { id: parsedId } });
+      if (fine == null) {
+        res.status(404).send();
+        return;
+      }
+
+      await DebtorService.deleteFine(parsedId);
+      res.status(204).send();
+    } catch (error) {
+      this.logger.error('Could not return fine handout event:', error);
+      res.status(500).json('Internal server error.');
+    }
+  }
+
+  /**
+   * Return all users that had at most -5 euros balance both now and on the reference date
+   * For all these users, also return their fine based on the reference date.
+   * @route GET /fines/eligible
+   * @group debtors - Operations of the debtor controller
+   * @security JWT
+   * @param {Array<string>} userTypes.query.required - List of all user types fines should be calculated for
+   * @param {string} referenceDate.query - Date to base fines on. If undefined, use now.
+   * @returns {Array<UserToFineResponse.model>} 200 - List of eligible fines
+   * @returns {string} 400 - Validation error
+   * @returns {string} 500 - Internal server error
+   */
+  public async calculateFines(req: RequestWithToken, res: Response): Promise<void> {
+    this.logger.trace('Get all possible fines by ', req.token.user);
+
+    let params;
+    try {
+      params = {
+        userTypes: asArrayOfUserTypes(req.query.userTypes),
+        referenceDate: asDate(req.query.referenceDate),
+      };
+      if (params.userTypes === undefined) throw new Error('userTypes is not a valid array of UserTypes');
+      if (params.referenceDate === undefined && req.query.referenceDate !== undefined) throw new Error('referenceDate is not a valid date');
+    } catch (e) {
+      res.status(400).send(e.message);
+      return;
+    }
+
+    try {
+      res.json(await DebtorService.calculateFinesOnDate(params));
+    } catch (error) {
+      this.logger.error('Could not calculate fines:', error);
+      res.status(500).json('Internal server error.');
+    }
+  }
+
+  /**
+   * Handout fines to all given users.
+   * @route POST /fines/handout
```

Comment:
mist unieke @operationId

Reply:
Resolved in cfcbd191b85b3707a4b3e2321c59065dc4a95ea6